data/customModules/Command.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * A Command Arg.
+ * @typedef  {Object}  Arg
+ * @property {String}  name         The name of the arg.
+ * @property {Boolean} [mand=false] Whether or not this arg is mandatory.
+ * @property {String}  [delim=' ']  The delimiting character for this arg.
+ */
+
+/**
+ * Class representing a command.
+ */
+class Command {
+  /**
+   * Create a command.
+   * @param {Object}   data                         The command data.
+   * @param {String}   data.name                    The command name.
+   * @param {String}   data.description             The command description.
+   * @param {Function} data.action                  The command action.
+   * @param {Object}   [data.options={}]            The command options.
+   * @param {Arg[]}    [data.options.arguments=[]]  List of arguments that the command takes.
```

Comment:
Maybe change this from `arguments` to just `args`?

Reply:
makes sense

Reply:
Oh, and change `description` to `desc`.